### PR TITLE
Implement asymptotic series expansion of Ci(x) at -oo

### LINKED ISSUE #26207: Asymptotic series expansion for Cosine Integral function
I was just testing out the asymptotic series & limits at infinity for some error functions and I found this 
```
>>> Si(x).series(x, oo)
pi/2 - (-6/x**3 + 1/x + O(x**(-7), (x, oo)))*sin(x)/x - (24/x**4 - 2/x**2 + 1 + O(x**(-7), (x, oo)))*cos(x)/x
>>>
>>> Si(x).series(x, -oo)
-24*cos(x)/x**5 + 6*sin(x)/x**4 + 2*cos(x)/x**3 - sin(x)/x**2 - cos(x)/x - pi/2 + O(x**(-6), (x, -oo))
>>>
>>> Ci(x).series(x, oo)
-(-6/x**3 + 1/x + O(x**(-7), (x, oo)))*cos(x)/x + (24/x**4 - 2/x**2 + 1 + O(x**(-7), (x, oo)))*sin(x)/x
>>>
>>> Ci(x).series(x, -oo)
Traceback (most recent call last):
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 2994, in series
    s = self.subs(x, cdir/x).series(x, n=n, dir='+', cdir=1)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3015, in series
    rv = self.subs(x, xpos).series(xpos, x0, n, dir, logx=logx, cdir=cdir)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3023, in series
    s1 = self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 1800, in _eval_nseries
    return super()._eval_nseries(x, n, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 690, in _eval_nseries
    return self._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 2082, in _eval_aseries
    return super(Ci, self)._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 653, in _eval_aseries
    raise PoleError(filldedent('''
sympy.core.function.PoleError:
Asymptotic expansion of Ci around [-oo] is not implemented.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 2999, in series
    s = self.subs(x, cdir*x).aseries(x, n=n)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3198, in aseries
    return self.subs(x, xpos).aseries(xpos, n, bound, hir).subs(xpos, x)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3216, in aseries
    s = self.subs(x, exp(x)).aseries(x, n, bound, hir).subs(x, log(x))
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3238, in aseries
    s = func.series(k, 0, n)
  File "C:\Users\91989\sympy\sympy\sympy\core\expr.py", line 3023, in series
    s1 = self._eval_nseries(x, n=n, logx=logx, cdir=cdir)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 1800, in _eval_nseries
    return super()._eval_nseries(x, n, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 690, in _eval_nseries
    return self._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\functions\special\error_functions.py", line 2082, in _eval_aseries
    return super(Ci, self)._eval_aseries(n, args0, x, logx)
  File "C:\Users\91989\sympy\sympy\sympy\core\function.py", line 653, in _eval_aseries
    raise PoleError(filldedent('''
sympy.core.function.PoleError:
Asymptotic expansion of Ci around [-oo] is not implemented.
```

### PR DESCRIPTION
**Pull request by mohakmalviya**
#### Brief description of what is fixed or changed
Implemented the asymptotic series expansion for the Cosine Integral function Ci(x) at negative infinity (-oo). Previously, attempting to compute this expansion would result in a PoleError due to the lack of implementation. This update provides a correct series expansion, including the necessary I*pi term, aligning with the mathematical expectations for Ci(x) as x approaches -oo.

#### Other comments
The implementation involves adjustments to the _eval_aseries method within the Ci class to handle expansions at -oo correctly. Tests have been added to ensure the accuracy of the expansion and its consistency with the expected mathematical behavior.

#### Release Notes

* functions
  * Implemented the asymptotic series expansion of Ci(x) at -oo, including the I*pi term

### CODE REVIEW

File: sympy/functions/special/error_functions.py [hunk 1]
```diff
@@ -2078,7 +2077,15 @@ def _eval_aseries(self, n, args0, x, logx):
...
+        # Expansion at -oo
+        elif point is S.NegativeInfinity:
+            z = self.args[0]
+            p = [S.NegativeOne**k * factorial(2*k) / z**(2*k)
+                    for k in range(int((n - 1)/2))] + [I*pi] + [Order(1/z**n, x)]
+            q = [S.NegativeOne**k * factorial(2*k + 1) / z**(2*k + 1)
+                    for k in range(int(n/2) - 1)] +[Order(1/z**n, x)]
+            return (sin(z)/z)*Add(*p) - (cos(z)/z)*Add(*q)
+
         return super(Ci, self)._eval_aseries(n, args0, x, logx)
```
Comment (anutosh491) [inline sympy/functions/special/error_functions.py:2089]: Thanks for this, I shall go through this soon. I am not sure what is the case with complex infinites like `I*oo`, give it a try if that interests you ;)

File: sympy/functions/special/tests/test_error_functions.py [hunk 2]
```diff
@@ -682,6 +682,10 @@ def test_ci():
     assert Ci(x).series(x, oo) == -cos(x)*(-6/x**3 + 1/x \
         + O(x**(-7), (x, oo)))/x + (24/x**4 - 2/x**2 + 1 \
         + O(x**(-7), (x, oo)))*sin(x)/x
+
+    assert Ci(x).series(x, -oo) == -cos(x)*(-6/x**3 + 1/x + O(-1/x**7, (x, -oo)))/x + \
+            (24/x**4 - 2/x**2 + I*pi + 1 + O(-1/x**7, (x, -oo)))*sin(x)/x
+
```
Comment (anutosh491) [inline sympy/functions/special/tests/test_error_functions.py:688]: Wait I don't think the `I*pi` term should be there inside this bracket `(24/x**4 - 2/x**2 + I*pi + 1 + O(-1/x**7, (x, -oo)))`
I think it should be `(24/x**4 - 2/x**2 + 1 + O(-1/x**7, (x, -oo)))*sin(x)/x + I*pi`

File: sympy/functions/special/error_functions.py [hunk 3]
```diff
@@ -2078,7 +2077,15 @@ def _eval_aseries(self, n, args0, x, logx):
...
-        # All other points are not handled
+        # Expansion at -oo
+        elif point is S.NegativeInfinity:
+            z = self.args[0]
+            p = [S.NegativeOne**k * factorial(2*k) / z**(2*k)
+                    for k in range(int((n - 1)/2))] + [I*pi] + [Order(1/z**n, x)]
+            q = [S.NegativeOne**k * factorial(2*k + 1) / z**(2*k + 1)
+                    for k in range(int(n/2) - 1)] +[Order(1/z**n, x)]
+            return (sin(z)/z)*Add(*p) - (cos(z)/z)*Add(*q)
+
```
Comment (anutosh491) [inline sympy/functions/special/error_functions.py:2088]: I think there is too much repetition of code here. 

Use this rather 
```
if point in (S.Infinity, S.NegativeInfinity):
     // write the code which i think is same in both cases
    // store in a variable like result
    if point is -oo:
        // Add the extract I*pi to result
    // return result
```

Hope that makes sense

File: sympy/functions/special/error_functions.py (hunk 1 above)
Comment (mohakmalviya) [reply sympy/functions/special/error_functions.py:2089]: sure sir!

File: sympy/functions/special/error_functions.py (hunk 3 above)
Comment (mohakmalviya) [reply sympy/functions/special/error_functions.py:2088]: okay sir

File: sympy/functions/special/error_functions.py [hunk 4]
```diff
@@ -1942,7 +1942,6 @@ def _eval_is_zero(self):
         if z.is_zero:
             return True
 
-
```
Comment (haru-44) [inline sympy/functions/special/error_functions.py:1945]: Deletion of this line is unnecessary.

**Review by haru-44 — DISMISSED** [review]

File: sympy/functions/special/error_functions.py (hunk 4 above)
Comment (mohakmalviya) [reply sympy/functions/special/error_functions.py:1945]: sir but it is not affecting the code and it was un necessary line

**Review by haru-44 — COMMENTED** [review]
The formal fix has been resolved, but I am unable to determine the mathematical content.

https://github.com/sympy/sympy/issues/26207#issuecomment-1949634664

**Comment by anutosh491** [comment]
Let's work in smaller bunch of changes if that makes sense @haru-44 @mohakmalviya 

To the above comment, I have replied on the issue (https://github.com/sympy/sympy/issues/26207#issuecomment-1951039015).
Let's get this in and then fix the number of terms required for asymptotic expansions for all error functions. Does that make sense ?

The changes here look good to get us started with an asymptotic expansion at `-oo`

**Review by anutosh491 — APPROVED** [review]

**Review by haru-44 — APPROVED** [review]

**Comment by anutosh491** [comment]
Merging this but I have updated the PR template and removed the issue number which was mentioned, as there are few things that should still be fixed through subsequent PRs. @mohakmalviya please go ahead and address those.

**Pull request merged**